src/libraries/System.Runtime.Loader/tests/LoaderLinkTest.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+using LoaderLinkTest.Shared;
+
+namespace LoaderLinkTest
+{
+    public class LoaderLinkTest
+    {
+        [Fact]
+        public static void EnsureTypesLinked() // https://github.com/dotnet/runtime/issues/42207
```

Comment:
Nit: I'm not very familiar with the language in the loader so feel free to ignore this. The current name sounds like it's talking about the trimmer \ linker. Should we call this `EnsureAssembliesAreLoadedFromTheDefaultLoadContext` or something along those lines?

Reply:
I think it's mostly fine? The idea is that we want to ensure you're referencing the same type, and 'linked' is a common way to refer to that.